Entity/Notification.php

```diff
@@ -133,5 +150,22 @@ public function setDelivered()
 
         $this->deliveredAt = new \DateTime('now');
     }
-}
 
+    /**
+     * @param NotificationAttachment[] $attachments
+     *
+     * @return $this
+     */
+    public function setAttachments($attachments)
+    {
+        $this->attachments = $attachments;
```

Comment:
`$this->attachents` internal type should stay `ArrayCollection` so think about clear the collection, iterate over `$attachments` and call `add` method